calyx/src/analysis/live_range_analysis.rs

```diff
@@ -9,44 +9,55 @@ use std::{
     ops::{BitOr, Sub},
 };
 
+type LiveCellRepresentation = HashSet<(ir::CellType, ir::Id)>;
+const NODE_ID: &str = "NODE_ID";
+
 /// The data structure used to represent sets of ids. This is used to represent
 /// the `live`, `gen`, and `kill` sets.
 #[derive(Default, Clone)]
 pub struct Prop {
-    set: HashSet<ir::Id>,
+    map: HashMap<ir::CellType, HashSet<ir::Id>>,
```

Comment:
Maybe we should define a local `type` alias for this `HashMap<...>` type? It occurs in a bunch of places so it'd be nice to know what it represents